File: minigit-5/ts/src/commands/rev-parse.ts
// rev-parse command - Resolve revisions

import { findRepoRoot } from '../utils';
import { resolveRevision } from '../refs';

export function revParse(args: string[]): number {
  const repoRoot = findRepoRoot();
  if (!repoRoot) {
    console.error('fatal: not a minigit repository');
    return 1;
  }

  // Parse arguments
  const revisions: string[] = [];

  for (const arg of args) {
    if (!arg.startsWith('-')) {
      revisions.push(arg);
    }
  }

  if (revisions.length === 0) {
    console.error('fatal: revision required');
    return 1;
  }

  for (const rev of revisions) {
    try {
      const sha = resolveRevision(rev, repoRoot);
      console.log(sha);
    } catch (e: any) {
      console.error(`fatal: ${e.message}`);
      return 1;
    }
  }

  return 0;
}

File: minigit-5/py/commands/rev_parse.py
"""rev-parse command - Resolve revisions to SHA."""

import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).parent.parent))

from utils import find_repo_root
from refs import resolve_revision


def run(args: list[str]) -> int:
    """Resolve revision to SHA."""
    repo_root = find_repo_root()
    if repo_root is None:
        print("fatal: not a minigit repository", file=sys.stderr)
        return 1

    if not args:
        print("fatal: revision required", file=sys.stderr)
        return 1

    revision = args[0]
    sha = resolve_revision(repo_root, revision)

    if sha is None:
        print(f"fatal: ambiguous argument '{revision}'", file=sys.stderr)
        return 1

    print(sha)
    return 0
